### ISSUE TITLE
VSCode (with the Python extension) runs commands with every single env var repeated in the shell command

### ISSUE THREAD

**Issue by ThiefMaster** [microsoft/vscode-remote-release#2017]
- VSCode Version: 1.41.0-insider
- Local OS Version: Win 10 1909 (18363.535)
- Remote OS Version: Gentoo Linux, Kernel 4.19.86, zsh 5.6.2
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Install the Python extension
2. Open some Python script (can be anything, even a simple hello world script)
3. Use CTRL-F5 (Run without Debug) twice

MANY more details on this issue are in my original issue, microsoft/vscode#85887, but I have been pointed to this repo so I'm reporting it here as well since it may be related to how the env vars are retrieved in remote editing mode.

**Comment by Tyriar** [comment]
@ThiefMaster doesn't this happen on non-remote as well when debugging?

**Comment by ThiefMaster** [comment]
Yes, the same problem happens locally (this was the output from using CTRL+F5 twice to run a hello-world python script)

![image](https://user-images.githubusercontent.com/179599/71527271-f95fad00-28da-11ea-8b73-aceb00424978.png)

**Comment by ThiefMaster** [comment]
When using a normal debug run (F5), it does not repeat a huge load of env vars:

![image](https://user-images.githubusercontent.com/179599/71527409-9e7a8580-28db-11ea-85ea-701a4a8f4461.png)

**Comment by Tyriar** [comment]
Moving debug discussion: https://github.com/microsoft/vscode/issues/87779
